nengo/cache.py

```diff
@@ -177,14 +181,7 @@ def __init__(self, readonly=False, cache_dir=None):
         self.cache_dir = cache_dir
         safe_makedirs(self.cache_dir)
         self._fragment_size = get_fragment_size(self.cache_dir)
-        try:
-            self._remove_legacy_files()
-            self._index = CacheIndex(os.path.join(self.cache_dir, self._INDEX))
-        except TimeoutError:
-            warnings.warn(
-                "Decoder cache could not acquire lock and was deactivated.")
-            self._index = {}
-            self.readonly = True
+        self._index = {}
```

Comment:
Why does this default to an empty dict, rather than say None? If it were None, then you could just check for that on `__exit__`, rather than having to use `hasattr`.